sdk/search/azure-search/src/main/java/com/azure/search/models/Field.java

```diff
@@ -447,12 +493,35 @@ public String getAnalyzer() {
      * the field. This option can be used only with searchable fields and it
      * can't be set together with either searchAnalyzer or indexAnalyzer. Once
      * the analyzer is chosen, it cannot be changed for the field. Must be null
-     * for complex fields.
+     * for complex fields. Possible values include: 'ar.microsoft',
+     * 'ar.lucene', 'hy.lucene', 'bn.microsoft', 'eu.lucene', 'bg.microsoft',
+     * 'bg.lucene', 'ca.microsoft', 'ca.lucene', 'zh-Hans.microsoft',
+     * 'zh-Hans.lucene', 'zh-Hant.microsoft', 'zh-Hant.lucene', 'hr.microsoft',
+     * 'cs.microsoft', 'cs.lucene', 'da.microsoft', 'da.lucene',
+     * 'nl.microsoft', 'nl.lucene', 'en.microsoft', 'en.lucene',
+     * 'et.microsoft', 'fi.microsoft', 'fi.lucene', 'fr.microsoft',
+     * 'fr.lucene', 'gl.lucene', 'de.microsoft', 'de.lucene', 'el.microsoft',
+     * 'el.lucene', 'gu.microsoft', 'he.microsoft', 'hi.microsoft',
+     * 'hi.lucene', 'hu.microsoft', 'hu.lucene', 'is.microsoft',
+     * 'id.microsoft', 'id.lucene', 'ga.lucene', 'it.microsoft', 'it.lucene',
+     * 'ja.microsoft', 'ja.lucene', 'kn.microsoft', 'ko.microsoft',
+     * 'ko.lucene', 'lv.microsoft', 'lv.lucene', 'lt.microsoft',
+     * 'ml.microsoft', 'ms.microsoft', 'mr.microsoft', 'nb.microsoft',
+     * 'no.lucene', 'fa.lucene', 'pl.microsoft', 'pl.lucene',
+     * 'pt-BR.microsoft', 'pt-BR.lucene', 'pt-PT.microsoft', 'pt-PT.lucene',
+     * 'pa.microsoft', 'ro.microsoft', 'ro.lucene', 'ru.microsoft',
+     * 'ru.lucene', 'sr-cyrillic.microsoft', 'sr-latin.microsoft',
+     * 'sk.microsoft', 'sl.microsoft', 'es.microsoft', 'es.lucene',
+     * 'sv.microsoft', 'sv.lucene', 'ta.microsoft', 'te.microsoft',
+     * 'th.microsoft', 'th.lucene', 'tr.microsoft', 'tr.lucene',
+     * 'uk.microsoft', 'ur.microsoft', 'vi.microsoft', 'standard.lucene',
+     * 'standardasciifolding.lucene', 'keyword', 'pattern', 'simple', 'stop',
+     * 'whitespace'.
      *
      * @param analyzer the analyzer value to set.
      * @return the Field object itself.
      */
-    public Field setAnalyzer(String analyzer) {
+    public Field setAnalyzer(AnalyzerName analyzer) {
```

Comment:
`indexAnalyzer` and `searchAnalyzer` should also be of type `AnalyzerName`. The Swagger is defined this way; I'm not sure why the generated code isn't.

Reply:
Swagger made the right return type for the three class.
In transformer, we covert the AnalyzerName Enum type to String. I only made the changes on the one on issue. Thanks for pointing this out. 
I will change them all.